content/frameworks/react.textile

```diff
@@ -0,0 +1,179 @@
+
+h2(#installation). Installation
+
+To install the Ably React Hooks, use the following command:
+
+```[sh]
+npm install --save ably
+```
+
+**Note:** You will need React 16.8 or above to use these hooks.
+
+h2(#set-up-ably). Set up Ably
+
+To utilize these hooks, you'll need an Ably API key. If you haven't signed up yet, sign up now for a free Ably account. Once registered:
+
+* Log into your app dashboard.
+* Under "Your apps", click on "Manage app" for any app you wish to use for this tutorial, or create a new one with the "Create New App" button.
+* Navigate to the "API Keys" tab.
+* Copy the secret "API Key" value from your Root key. This will be used later when building your app.
+
+**Note:** It's recommended to use Token Authentication. This requires server-side code, which is beyond the scope of this documentation. In the examples provided, we use an API key directly in the markup. This approach is for local development only and should not be used in production. Ensure you don't commit this to your repositories.
+
+h2(#usage). Usage
+
+h3(#ably-provider). Setting up the Ably Provider
+
+To connect your app to Ably, use the @AblyProvider@ component. This component should be placed high in your component tree, wrapping every component that needs to access Ably.
+You can create your own client and pass it to the context provider:
+
+```[react]
+import { Realtime } from "ably";
+import { AblyProvider } from 'ably/react';
+
+const client = new Realtime({ key: "your-ably-api-key", clientId: 'your-ably-client-id' });
+
+root.render(
+  <AblyProvider client={client}>
+    <App />
+  </AblyProvider>
+);
+```
+
+Alternatively, you can create your options and pass them to the context provider:
+
+```[react]
+import * as Ably from 'ably';
+import { AblyProvider } from 'ably/react';
+
+const options = {
+  key: "your-ably-api-key",
+  clientId: "your-ably-client-id",
+}
+
+root.render(
+  <AblyProvider options={options}>
+    <App />
+  </AblyProvider>
+);
+```
+
+h3(#useChannel). useChannel
+
+The @useChannel@ hook allows you to subscribe to a channel and receive its messages. It can be combined with the React @useState@ hook to maintain a list of messages in your app's state.
+
+```[react]
+const [messages, updateMessages] = useState([]);
+const { channel } = useChannel("your-channel-name", (message) => {
+    updateMessages((prev) => [...prev, message]);
+});
+```
+
+You can also filter messages by providing a message type to the @useChannel@ function:
+
+```[react]
+const { channel } = useChannel("your-channel-name", "test-message", (message) => {
+    console.log(message);
+});
+```
+
+h3(#usePresence). usePresence
+
+The @usePresence@ hook allows you to subscribe to presence events on a channel, notifying you when a user joins or leaves.
+
+```[react]
+const { presenceData, updateStatus } = usePresence("your-channel-name");
```

Comment:
maybe add example with `updateStatus`? maybe with `useEffect`

Reply:
That's a good point! I could maybe convert this to something simple like:

```
const { presenceData, updateStatus } = usePresence("your-channel-name", "initial state");

// Convert presence data to list items to render    
const peers = presenceData.map((msg, index) => <li key={index}>{msg.clientId}: {msg.data}</li>);

// The `updateStatus` function can be used to update the presence data for the current client
updateStatus("new status");
```

Or did you have something else in mind?